pkg/sql/codegen/tensorflow/template_pred.go

```diff
@@ -78,148 +47,29 @@ feature_metas["{{$value.Name}}"] = {
-        dataset = dataset.map(ds_mapper).batch(1)
-        iterator = dataset.make_one_shot_iterator()
-        features = iterator.get_next()
-        return features
-
-    return _inner_input_fn
-
-class FastPredict:
-    def __init__(self, estimator, input_fn):
-        self.estimator = estimator
-        self.first_run = True
-        self.closed = False
-        self.input_fn = input_fn
-
-    def _create_generator(self):
-        while not self.closed:
-            yield self.next_features[0], self.next_features[1]
-
-    def predict(self, feature_batch):
-        self.next_features = feature_batch
-        if self.first_run:
-            self.batch_size = len(feature_batch)
-            self.predictions = self.estimator.predict(
-                input_fn=self.input_fn(self._create_generator))
-            self.first_run = False
-        elif self.batch_size != len(feature_batch):
-            raise ValueError("All batches must be of the same size. First-batch:" + str(self.batch_size) + " This-batch:" + str(len(feature_batch)))
-
-        results = []
-        for _ in range(self.batch_size):
-            results.append(next(self.predictions))
-        return results
-
-    def close(self):
-        self.closed = True
-        try:
-            next(self.predictions)
-        except Exception as e:
-            print("Exception in fast_predict. This is probably OK: %s" % e)
-
-column_names = feature_column_names[:]
-column_names.append("{{.Y.Name}}")
-pred_gen = db_generator(driver, conn, """{{.Select}}""", feature_column_names, "{{.Y.Name}}", feature_metas)()
-fast_predictor = FastPredict(classifier, fast_input_fn)
-
-with buffered_db_writer(driver, conn, "{{.ResultTable}}", column_names, 100) as w:
-    while True:
-        try:
-            features = pred_gen.__next__()
-        except StopIteration:
-            break
-        result = fast_predictor.predict(features)
-        row = []
-        for idx, _ in enumerate(feature_column_names):
-            val = features[0][idx]
-            row.append(str(val))
-        if "class_ids" in list(result)[0]:
-            row.append(str(list(result)[0]["class_ids"][0]))
-        else:
-            # regression predictions
-            row.append(str(list(result)[0]["predictions"][0]))
-        w.write(row)
+label_meta = {
+    "feature_name": "{{.Y.Name}}",
+    "dtype": "{{.Y.DType | dtypeToString}}",
+    "delimiter": "{{.Y.Delimiter}}",
+    "shape": {{.Y.Shape | intArrayToJSONString}},
+    "is_sparse": "{{.Y.IsSparse}}" == "true"
+}
 
+model_params=dict()
+{{range $k, $v := .ModelParams}}
+model_params["{{$k}}"]={{$v | attrToPythonValue}}
 {{end}}
 
-print("Done predicting. Predict table : {{.ResultTable}}")
+pred(is_keras_model="{{.IsKerasModel}}" == "true",
+    datasource="{{.DataSource}}",
+    estimator="{{.Estimator}}",
+    select="""{{.Select}}""",
+    result_table="{{.ResultTable}}",
+    feature_column_code="""{{.FeatureColumnCode}}""",
```

Comment:
Is it possible to avoid `feature_column_code` and `estimator_code`? So that we don't need to call `eval`.

Reply:
Sounds good, will try to refine this in next PR.